### ISSUE TITLE
serial poll() is always reading only 1 byte per call

### ISSUE THREAD

**Issue by ku3i** [arduino-libraries/Arduino_MKRGPS#35]
https://github.com/arduino-libraries/Arduino_MKRGPS/blob/c736985026d5b72514faec561c912c2035b0b801/src/GPS.cpp#L195

The current version is "taking forever" to read a GPS location, if the user has a significant delay (e.g. 500ms) in the loop() method, since on each GPS.available() call only 1 byte is read.

proposed fix: replace the "if" with a "while" in line 195 (GPS.cpp) 

Another tip: make the GPS_DEBUG flag available in the begin() method (2nd argument or so), to better track issues like this easily and without modifying the library.

**Comment by AlexBimmerHead** [comment]
Yes i have modified the library this way and work perfectly with a delay of 300ms on the loop which read GPS data.
